docs/en/sql-reference/aggregate-functions/grouping_function.md

```diff
@@ -0,0 +1,640 @@
+GROUP BY
+    GROUPING SETS (
+        (number),
+        (number % 2)
+    )
+ORDER BY number, gr;
+```
+
+Response
+```response
+┌─number─┬─gr─┐
+│      0 │  0 │
+└────────┴────┘
+┌─number─┬─gr─┐
+│      0 │  1 │
+│      0 │  1 │
+└────────┴────┘
+┌─number─┬─gr─┐
+│      1 │  0 │
+│      2 │  0 │
+│      3 │  0 │
+│      4 │  0 │
+│      5 │  0 │
+│      6 │  0 │
+│      7 │  0 │
+│      8 │  0 │
+│      9 │  0 │
+└────────┴────┘
+
+12 rows in set. Elapsed: 0.004 sec.
+```
+
+```sql
+SELECT
+    number
+FROM numbers(10)
+GROUP BY
+    GROUPING SETS (
+        (number),
+        (number % 2)
+    )
+ORDER BY number, grouping(number, number % 2) = 1;
+```
+
+Response
+```response
+┌─number─┐
+│      0 │
+└────────┘
+┌─number─┐
+│      0 │
+│      0 │
+└────────┘
+┌─number─┐
+│      1 │
+│      2 │
+│      3 │
+│      4 │
+│      5 │
+│      6 │
+│      7 │
+│      8 │
+│      9 │
+└────────┘
+
+12 rows in set. Elapsed: 0.002 sec.
+```
+
+```sql
+SELECT
+    number,
+    count(),
+    grouping(number, number % 2) AS gr
+FROM numbers(10)
+GROUP BY
+    GROUPING SETS (
+        (number),
+        (number, number % 2),
+        ()
+    )
+ORDER BY (gr, number);
+```
+
+Response
+```response
+┌─number─┬─count()─┬─gr─┐
+│      0 │      10 │  0 │
+└────────┴─────────┴────┘
+┌─number─┬─count()─┬─gr─┐
+│      0 │       1 │  2 │
+│      1 │       1 │  2 │
+│      2 │       1 │  2 │
+│      3 │       1 │  2 │
+│      4 │       1 │  2 │
+│      5 │       1 │  2 │
+│      6 │       1 │  2 │
+│      7 │       1 │  2 │
+│      8 │       1 │  2 │
+│      9 │       1 │  2 │
+└────────┴─────────┴────┘
+┌─number─┬─count()─┬─gr─┐
+│      0 │       1 │  3 │
+│      1 │       1 │  3 │
+│      2 │       1 │  3 │
+│      3 │       1 │  3 │
+│      4 │       1 │  3 │
+│      5 │       1 │  3 │
+│      6 │       1 │  3 │
+│      7 │       1 │  3 │
+│      8 │       1 │  3 │
+│      9 │       1 │  3 │
+└────────┴─────────┴────┘
+
+21 rows in set. Elapsed: 0.015 sec.
+```
+
+```sql
+SELECT
+    number
+FROM numbers(10)
+GROUP BY
+    GROUPING SETS (
+        (number),
+        (number % 2)
+    )
+HAVING grouping(number, number % 2) = 2
+ORDER BY number
+```
+
+Response
+```response
+Received exception from server (version 22.7.1):
+Code: 47. DB::Exception: Received from localhost:9000. DB::Exception: Unknown identifier: __grouping_set: While processing grouping(number, number % 2) = 2. (UNKNOWN_IDENTIFIER)
+```
+
+```sql
+SELECT
+    number
+FROM numbers(10)
+GROUP BY
+    GROUPING SETS (
+        (number),
+        (number % 2)
+    )
+HAVING grouping(number, number % 2) = 1
+ORDER BY number
+```
+
+Response
+```response
+Received exception from server (version 22.7.1):
+Code: 47. DB::Exception: Received from localhost:9000. DB::Exception: Unknown identifier: __grouping_set: While processing grouping(number, number % 2) = 1. (UNKNOWN_IDENTIFIER)
+```
+
+```sql
+SELECT
+    number,
+    GROUPING(number, number % 2) = 1 as gr
+FROM remote('172.21.0.{2,4}', numbers(10))
```

Comment:
I'd like to see real-world examples instead of equilibristics with system.numbers.